### PR TITLE
[cmake build]

### PR DESCRIPTION
**Pull request by rhilkens**
[Change]
- introduce CMAKE build for MLI
- allows to compile on MSVC (X86 Windows),
  GCC/LLVM (X86 Linux) and MWDT.

### CODE REVIEW

File: examples/hello_world/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Copyright 2020, Synopsys, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-3-Clause license found in
+# the LICENSE file in the root directory of this source tree.
+#
+
+include(../../cmake/settings.cmake)
+
+add_executable(MLI_unit_test
```
Comment (dzakhar) [inline examples/hello_world/CMakeLists.txt:11]: Why it's name MLI_Unit_test?

File: examples/hello_world/main.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
+* Copyright 2020, Synopsys, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the BSD-3-Clause license found in
+* the LICENSE file in the root directory of this source tree.
+*
+*/
+#include "mli_api.h"
+#include "mli_check.h"
```
Comment (dzakhar) [inline examples/hello_world/main.cpp:10]: mli check not a public header. What's the purpose to have it here?

File: examples/hello_world/main.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    int8_t data_out[8] = { 1, 2, 3, 4, 5, 6, 7, 8 };
+    mli_tensor out = { 0 };
+    out.el_type = MLI_EL_FX_8;
+    out.rank = 1;
+    out.shape[0] = 8;
+    out.mem_stride[0] = 1;
+    out.capacity = sizeof(data_out);
+    out.data = data_out;
+    mli_status status;
```
Comment (dzakhar) [inline examples/hello_world/main.cpp:34]: Can it be declare below at line 36? or at least initialized here?

File: examples/hello_world/make/Makefile [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ifeq ($(UNAME),Windows)
+TOOL_CHAIN_OPTIONS = -AWin32
+else
+TOOL_CHAIN_OPTIONS =
+endif
+
+else
+# Set target specific setting for ARC build
+ifeq ($(TARGET),ARC)
+BUILD_FOLDER = ../../../outputs_mwdt
```
Comment (dzakhar) [inline examples/hello_world/make/Makefile:39]: There is a chance that result project can use ARC GNU as toolchain. In this context I think it's confusing to stick to *_mwdt postfix

File: lib/src/pal/ref/mli_math.h [hunk 5]
```diff
@@ -25,6 +25,16 @@ MLI_FORCE_INLINE T mli_math_limit_fx(T sign) {
     return sign < (T)0 ? std::numeric_limits<T>::lowest() : std::numeric_limits<T>::max();
 }
 
+template <typename T>
+MLI_FORCE_INLINE T mli_math_asr_pos_nbits_fx(T x, int nbits)
+{
+    // This function is internal to this file
+    // and is required to avoid compiler issues.
+    if (nbits > (sizeof(T) * 8 - 1))
```
Comment (dzakhar) [inline lib/src/pal/ref/mli_math.h:33]: mli_math_asr allows nbits to be negative. I don't see how it is handeled here.

**Review by dzakhar — CHANGES_REQUESTED** [review]

File: lib/src/pal/ref/mli_math.h (hunk 5 above)
Comment (JaccovG) [reply lib/src/pal/ref/mli_math.h:33]: This function is only called for positive nbits.

**Review by JaccovG — APPROVED** [review]

**Pull request merged**